### PR TITLE
Rewrite build system and development scripts

### PR DESCRIPTION
**Pull request by LukasKalbertodt**
This PR touches quite a few files, but I think it's a big improvement. I replaced `floof.yaml` by a number of bash scripts that offer the same features.

Main advantages/reasons for this change:

- Remove dependency on `floof`, a project that I have given up on (it's too complex to fit in my time budget for personal projects)
- Remove dependency on `watchman`: this is required by `relay-compiler --watch` and is a pain in the but to install systems for which there is no prebuilt binary. (Getting rid of that had nothing to do with getting rid of floof.)
- The docker containers are now all defined in a single `docker-compose` file, which is way more idiomatic I think.

There are plenty of other small improvements. See commit messages for more details. However, I would recommend just reviewing the final state of everything.

### CODE REVIEW

**Comment by LukasKalbertodt** [comment]
Note: test deployment does not work due to the CI scripts being taken from `master` instead of this branch.

**Review by JulianKniephoff — CHANGES_REQUESTED** [review]

File: x.sh [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            "$basedir"/util/scripts/check-system.sh building-only
+            >&2 echo
+            >&2 echo "Your system is missing some tools. Run './x.sh check-system' to repeat this check."
+            exit 1
+        fi
+        "$basedir"/util/scripts/build-release.sh
+        ;;
+    "containers")
+        containers "$2"
+        ;;
```
Comment (JulianKniephoff) [inline x.sh:88]: I find it a bit strange that there is no explicit script for this.

**Comment by JulianKniephoff** [comment]
What about scripts for reseeding the database? :thinking:

**Comment by JulianKniephoff** [comment]
In a first test run everything seemed to work nicely, though. After some fiddling with Docker which might not have anything to do with this change. :shrug:

File: x.sh [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+# Manage the dev containers
+containers() {
+    if ! command -v docker-compose &> /dev/null; then
+        >&2 echo "'docker-compose' is not installed! (Also see './x.sh check-system')"
+        exit 1
+    fi
+
+    docker_command="docker-compose -f $basedir/util/containers/docker-compose.yml"
+    case "$1" in
```
[the remark this answers is not in the document]
Comment (LukasKalbertodt) [reply x.sh:41]: I don't think either of us needs it now and it's easy to do that without our script. So :shrug:  not now

File: x.sh (hunk 1 above)
Comment (LukasKalbertodt) [reply x.sh:88]: Yeah fair point. But currently the function talks explicitly about `x.sh` as it also does some checking. Moving that to a separate script and removing the reference to `x.sh` would make the error message worse. Just moving the actual executing to a script seems not necessary as it is so little? Dunno, I see your point, but it doesn't bother me and I would leave it.

File: x.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (JulianKniephoff) [reply x.sh:41]: OK but prepare for a PR of mine adding it soon(TM). xP

**Pull request merged**